postprocessor/src/test/java/com/azure/autorest/postprocessor/util/PartialUpdateHandlerTest.java

```diff
@@ -136,4 +136,63 @@ public void testGeneratedFileRemoveAPIAndExistingFileUpdateMethod() throws URISy
         assertEquals("test", compilationUnit.getTypes().get(0).getMethodsByName("putNullWithResponse").get(0).getParameter(1).getName().asString());
     }
 
+    @Test
+    public void testModuleInfoFile_WhenGeneratedFileEqualsExistingFile_ThenUseGeneratedFile() {
+        String existingFileContent = "// Copyright (c) Microsoft Corporation. All rights reserved.\n" +
+                "// Licensed under the MIT License.\n" +
+                "// Code generated by Microsoft (R) AutoRest Code Generator.\n" +
+                "\n" +
+                "module com.azure.iot.deviceupdate {\n" +
+                "\n" +
+                "    requires transitive com.azure.core;\n" +
+                "\n" +
+                "    exports com.azure.iot.deviceupdate;\n" +
+                "}\n";
+        String generatedFileContent = "// Copyright (c) Microsoft Corporation. All rights reserved.\n" +
+                "// Licensed under the MIT License.\n" +
+                "// Code generated by Microsoft (R) AutoRest Code Generator.\n" +
+                "\n" +
+                "module com.azure.iot.deviceupdate {\n" +
+                "\n" +
+                "    requires transitive com.azure.core;\n" +
+                "\n" +
+                "    exports com.azure.iot.deviceupdate;\n" +
+                "}\n";
+
+        String output = PartialUpdateHandler.handlePartialUpdateForFile(generatedFileContent, existingFileContent);
+
+        CompilationUnit compilationUnit = parse(output);
+        assertEquals(true, compilationUnit.getModule().isPresent());
+        assertEquals("com.azure.iot.deviceupdate", compilationUnit.getModule().get().getName().toString());
+        assertEquals(2, compilationUnit.getModule().get().getDirectives().size());
+    }
+
+    @Test
+    public void testModuleInfoFile_WhenExistingFileHasUpdates_ThenUseExistingFile() {
+        String existingFileContent = "// Copyright (c) Microsoft Corporation. All rights reserved.\n" +
+                "// Licensed under the MIT License.\n" +
+                "// Code generated by Microsoft (R) AutoRest Code Generator.\n" +
+                "module com.azure.messaging.webpubsubnew {\n" +
+                "    requires transitive com.azure.core;\n" +
+                "\n" +
+                "    exports com.azure.messaging.webpubsubnew;\n" +
+                "    exports com.azure.messaging.webpubsubnew.models;\n" +
+                "}";
+        String generatedFileContent = "// Copyright (c) Microsoft Corporation. All rights reserved.\n" +
+                "// Licensed under the MIT License.\n" +
+                "// Code generated by Microsoft (R) AutoRest Code Generator.\n" +
+                "module com.azure.messaging.webpubsubnew {\n" +
```

Comment:
Maybe change the module name, so that the assert afterwards will verify it?

Reply:
This is a real scenario from Webpubsub, they add another directive, but don't change module name, so I want to simulate their use case here.